spec/facility_spec.rb

```diff
@@ -23,4 +37,489 @@
+    end
+
+    it 'plate_type method returns correct plate for ev' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility_1.registered_vehicles).to eq([@bolt])
+    end
+  end
+
+  describe '#collected_fees' do
+    it 'collected_fees returns correct value for regular' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+
+      expect(@facility_1.collected_fees).to eq(100)
+    end
+
+    it 'collected_fees returns correct value for antique' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@camaro)
+
+      expect(@facility_1.collected_fees).to eq(25)
+    end
+
+    it 'collected_fees returns correct value for ev' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@facility_1.collected_fees).to eq(200)
+    end
+  end
+
+  describe 'registered_vehicles final' do
+    it 'facility will have all three vehicles registered.' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      @facility_1.register_vehicle(@camaro)
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+    end
+  end
+
+  describe 'collected_fees final' do
+    it 'facility will have collected fees for all vehicles' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      @facility_1.register_vehicle(@camaro)
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+  end
+
+  describe 'facility.services' do
+    it 'facilities without "Vehicle Registration" service cannot register vehicles' do
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.services).to eq([])
+      expect(@facility_2.register_vehicle(@bolt)).to eq(nil)
+    end
+  end
+
+  # Start 'Getting a Driver's License' section of Iteration 2
+  #  Administer a written test
+  # A written test can only be administered to registrants with a permit and
+  # who are at least 16 years of age
+
+  describe 'new registrants' do
+    before(:each) do
+      @registrant_1 = Registrant.new('Bruce', 18, true)
+      @registrant_2 = Registrant.new('Penny', 16)
+      @registrant_3 = Registrant.new('Tucker', 15)
+    end
+
+    it 'registrants license data is correct' do
+      expect(@registrant_1.license_data).to eq({ written: false, license: false, renewed: false })
+      expect(@registrant_2.license_data).to eq({ written: false, license: false, renewed: false })
+      expect(@registrant_3.license_data).to eq({ written: false, license: false, renewed: false })
+    end
+
+    it 'registrants permit data is correct (tests permit? method)' do
+      expect(@registrant_1.permit?).to eq(true)
```

Comment:
This test and the one above it are not necessary since you've already tested these methods in the Registrant spec class, so we don't need to retest them here. We can assume they are working